src/libraries/System.Text.Json/gen/JsonSourceGenerator.Parser.cs

```diff
@@ -1588,25 +1526,29 @@ SpecialType.System_String or
                     _builtInSupportTypes.Contains(type);
             }
 
-            private void PopulateBuiltInSupportTypes()
+            private static HashSet<ITypeSymbol> CreateBuiltInSupportTypeSet(KnownTypeSymbols knownSymbols)
             {
-                HashSet<ITypeSymbol> builtInSupportTypes = _builtInSupportTypes;
-
-                AddTypeIfNotNull(_knownSymbols.ByteArrayType);
-                AddTypeIfNotNull(_knownSymbols.TimeSpanType);
-                AddTypeIfNotNull(_knownSymbols.DateTimeOffsetType);
-                AddTypeIfNotNull(_knownSymbols.DateOnlyType);
-                AddTypeIfNotNull(_knownSymbols.TimeOnlyType);
-                AddTypeIfNotNull(_knownSymbols.GuidType);
-                AddTypeIfNotNull(_knownSymbols.UriType);
-                AddTypeIfNotNull(_knownSymbols.VersionType);
-
-                AddTypeIfNotNull(_knownSymbols.JsonArrayType);
-                AddTypeIfNotNull(_knownSymbols.JsonElementType);
-                AddTypeIfNotNull(_knownSymbols.JsonNodeType);
-                AddTypeIfNotNull(_knownSymbols.JsonObjectType);
-                AddTypeIfNotNull(_knownSymbols.JsonValueType);
-                AddTypeIfNotNull(_knownSymbols.JsonDocumentType);
+#pragma warning disable RS1024 // Compare symbols correctly https://github.com/dotnet/roslyn-analyzers/issues/5804
```

Comment:
This issue was closed over a year ago.  Is it still relevant?

Reply:
cc @ViktorHofer

Reply:
Unsure if that's worth it, given that we want to remove the roslyn3.11 generators at some point from our packages.

Reply:
This false positive will still be an issue for 4.0.1.
The updated transitive dependency is in 4.1.0

Reply:
We will likely be able to upgrade roslyn4.0 to a higher in support version in addition to removing roslyn3.11. That said, this isn't yet committed so I hesitate sharing details at this point.